### PR TITLE
`PrepSelPrep` template works with `torch`

### PR DESCRIPTION
**Pull request by willjmax**
This PR fixes bug #6185

### CODE REVIEW

File: pennylane/templates/subroutines/prepselprep.py [hunk 1]
```diff
@@ -23,22 +23,16 @@
...
 def _get_new_terms(lcu):
     """Compute a new sum of unitaries with positive coefficients"""
-
-    new_coeffs = []
+    coeffs, ops = lcu.terms()
     new_ops = []
 
-    for coeff, op in zip(*lcu.terms()):
-
+    for coeff, op in zip(coeffs, ops):
```
Comment (obliviateandsurrender) [inline pennylane/templates/subroutines/prepselprep.py:29]: Maybe we can compute `angles = qml.math.angle(coeffs)` outside the loop and then zip over `angles` and `ops`.

**Review by obliviateandsurrender — COMMENTED** [review]
Thanks @willjmax! Looks good, just one minor suggestion.

**Review by KetpuntoG — COMMENTED** [review]
Looks good! 😊 Remember to add an entry in `pennylane/doc/releases/changelog-dev.md` mentioning the bug and adding your name at the end of the file. 

On the other hand I noticed that you have forked instead of cloned [not a blocker]. Let me know if you don't have write access to the repository and I'll ask for it

**Review by KetpuntoG — APPROVED** [review]
Looks great! Thanks for the fix 🚀

File: doc/releases/changelog-dev.md [hunk 2]
```diff
@@ -21,6 +21,9 @@
 * Fix Pytree serialization of operators with empty shot vectors:
   [(#6155)](https://github.com/PennyLaneAI/pennylane/pull/6155)
 
+* Fix PrepSelPrep template to work with torch:
+  [(#6191)](https://github.com/PennyLaneAI/pennylane/pull/6191)
+
 <h3>Contributors ✍️</h3>
 
 This release contains contributions from (in alphabetical order):
```
Comment (obliviateandsurrender) [inline doc/releases/changelog-dev.md:29]: Please add your name to the contribution list. (alphabetical order of the last name)

**Review by obliviateandsurrender — APPROVED** [review]

File: doc/releases/changelog-dev.md [hunk 3]
```diff
@@ -21,6 +21,9 @@
 * Fix Pytree serialization of operators with empty shot vectors:
   [(#6155)](https://github.com/PennyLaneAI/pennylane/pull/6155)
 
+* Fix PrepSelPrep template to work with torch:
```
Comment (obliviateandsurrender) [inline doc/releases/changelog-dev.md:24]:
```suggestion
* Fix `qml.PrepSelPrep` template to work with `torch`:
```

**Comment by KetpuntoG** [comment]
Awesome! Feel free to merge

**Pull request merged**